docs/architecture/matching-service.md

```diff
@@ -5,9 +5,15 @@
 <!-- https://lucid.app/lucidchart/0202e4b8-5258-4cd6-a6a0-67159300532b/edit -->
 <img src="../_assets/matching-context.svg">
 
+## Task Queues
 Matching Service instances manage [Task Queues](https://docs.temporal.io/workers#task-queue) being polled by Temporal Worker processes.
 Long-poll requests from Temporal Workers are received by the Frontend Service, which routes them to the Matching Service instance responsible for the requested Task Queue.
 The Matching Service instance responds by sending Workflow Tasks and Activity Tasks from the requested Task Queue.
 A single Task Queue is responsible for delivering tasks relating to many Workflow Executions.
 
-Documentation of Matching Service internals is not yet available.
+### Task Queue Partitions
+Matching Service splits Task Queues into partitions to provide higher throughput overall. Default is 4, but there can be more. Partitions can be redistributed, loaded from storage, and unloaded to storage. Unloading/loading will preserve any existing backlog of tasks in a TaskQueue.
+
+When throughput of tasks is low or polling by workers is rare, a polling worker can be "sync-matched" from an empty partition to a parent partition. This is also true for a task on a partition which is not being polled, the task will attempt a "sync match" with a parent partition, hoping to find a poller. Typically a Task Queue will have a single parent, the root partition, but Task Queues with more partitions can have layers of parents, eventually all converging at the root partition. If a root partition of a Task Queue is loaded from storage, this will force all other partitions of that Task Queu to also load. This ensures that a "sync-match" can occur between a child partition and a long-awaited poller.
```

Comment:
that's not what sync match means, you're talking about "forwarding"

Reply:
Good to know. What is a Sync match then? could this be old terminology? I noticed the term in comments.

Reply:
Sync match is when a task is matched with a poller without ever being persisted to the db backlog. If there is no (worker) poller available, the task is persisted to the db backlog. When a task is read from the backlog and then matched with a poller, it's called an async match